src/transform/src/reduce_elision.rs

```diff
@@ -35,48 +36,83 @@ impl crate::Transform for ReduceElision {
         relation: &mut MirRelationExpr,
         _: &mut TransformCtx,
     ) -> Result<(), crate::TransformError> {
-        let result = relation.try_visit_mut_post(&mut |e| self.action(e));
+        // Assemble type information once for the whole expression.
+        let mut builder = DerivedBuilder::default();
+        builder.require::<RelationType>();
+        builder.require::<UniqueKeys>();
+        let derived = builder.visit(relation);
+        let derived_view = derived.as_view();
```

Comment:
Would it make sense to only try to gather this information if `relation` is `Reduce`? Otherwise, it seems we'd compute this but then never use it. Otoh, I'm not familiar enough with the transform infra to know how frequently we'll actually call it.

Reply:
So, we are at the point where we are going to compute or not the thing for the whole AST. You are right that we could traverse the AST and if there is no `Reduce` then .. return early. But we would need it even if `relation` is not a `Reduce` because it could have children that are.